### PR TITLE
check current execution segment on null

### PR DESCRIPTION
**Pull request by vhatsura**
Check the current execution segment on null before using it.

Addresses #4

### CODE REVIEW

File: src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs [hunk 1]
```diff
@@ -29,8 +31,10 @@ internal class HotChocolateDiagnosticListener : DiagnosticEventListener
 
         public override IActivityScope ExecuteRequest(IRequestContext context)
         {
-            ITransaction transaction = _apmAgent.Tracer.CurrentTransaction;
-            return new RequestActivityScope(context, transaction, _apmAgent);
+            ITransaction? transaction = _apmAgent.Tracer.CurrentTransaction;
+            return transaction != null
+                ? new RequestActivityScope(context, transaction, _apmAgent)
+                : _emptyActivityScope;
```
Comment (glucaci) [inline src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs]: use EmptyScope property from Base class

File: src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ internal class HotChocolateDiagnosticListener : DiagnosticEventListener
         private readonly IConfigurationReader _configuration;
         private readonly IApmLogger _apmLogger;
 
+        private readonly IActivityScope _emptyActivityScope = new EmptyActivityScope();
```
Comment (glucaci) [inline src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs]: remove and use EmptyScope property from Base class

**Review by glucaci — CHANGES_REQUESTED** [review]
Thanks, look good, please change the empty activity.

File: src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs (hunk 1 above)
Comment (vhatsura) [reply src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs]: ops, didn't see that there is an option to use it. Fixed

File: src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs (hunk 2 above)
Comment (vhatsura) [reply src/Elastic.Apm.GraphQL.HotChocolate/HotChocolateDiagnosticListener.cs]: Fixed

**Review by glucaci — CHANGES_REQUESTED** [review]
Please remove also the EmptyActivityScope class and the file itself :), than I will merge it and do a new release ;)

**Review by glucaci — APPROVED** [review]

**Pull request merged**